packages/hubble-sdk/src/Hubble.ts

```diff
@@ -328,7 +330,30 @@ export class Hubble {
         `Could not get HBB circulating supply from the HBB mint account: ${this._config.borrowing.accounts.mint.HBB}`
       );
     }
-    return new Decimal(tokenSupply.value.uiAmountString);
+    let totalTokenSupply = new Decimal(tokenSupply.value.uiAmountString);
+
+    if (this._cluster === 'mainnet-beta') {
+      try {
+        const streams = await Stream.get({
+          connection: this._connection,
+          wallet: new PublicKey(STREAMFLOW_HBB_CONTRACT),
+          cluster: Cluster.Mainnet,
+        });
+        let notWithdrawnTokens = new Decimal(0);
+        for (let [pubkey, stream] of streams) {
+          const totalWithdrawn = new Decimal(stream.withdrawnAmount.toString()).add(
+            stream.streamflowFeeWithdrawn.toString()
+          );
+          const deposited = new Decimal(stream.depositedAmount.toString());
+          notWithdrawnTokens = notWithdrawnTokens.add(deposited.minus(totalWithdrawn).dividedBy(HBB_DECIMALS));
+        }
+        totalTokenSupply = totalTokenSupply.minus(notWithdrawnTokens);
+      } catch (exception) {
+        throw Error(`Could not get HBB Streamflow contract data: ${exception}`);
```

Comment:
no throw please, this is a public api, find another way

Reply:
This is the SDK, not the API. I believe the applications using our SDK should be the ones handling these errors.